### DISCUSSION TITLE
Is invalidating all queries possible?

### DISCUSSION THREAD

**Discussion by hmbrg** [TanStack/query#1588] [Q&A] [answered]
I'm trying to invalidate all my queries at once. I know that sounds weird, but the use case is that our APIs work with and without authorization, but you get user specific data when logged in. 

I tried `queryCache.refetchQueries({ active: true });` and `queryCache.invalidateQueries({ active: true });` but both sadly don't seem to work. When I specify a specific query everything works as intended.

According to the docs, the query key should be optional on both methods, but that's not the behavior I'm seeing. Is this invalidating all queries support?

**Comment by dusty** [comment]
I think `.clear` should achieve what you want.

https://react-query.tanstack.com/reference/QueryCache#querycacheclear

**Reply by hmbrg** [reply]
Sadly, this doesn't seem to work for me.

**Reply by ansh** [reply]
Answer is queryClient.invalidateQueries() as given below by @obsd

**Comment by TkDodo** [comment]
Can you create a codesandbox? I think this should work...

**Reply by carlosriveroib** [reply]
I think isn't working for me too, queryCache.clear() doesn't rerender.

I have this Provider

```javascript
function UserProvider({ children, isAdmin, routeProtected }: UserProviderProps) {
  const { data, isLoading } = useQuery(
    ['me'],
    async () => {
      const response = await axios.get<User>('/api/user');

      return response.data;
    });

  console.log('dataProvider: ', data);

  if (routeProtected && !data && isLoading) {
    return <h1>Loading...</h1>;
  }

  if (routeProtected && !isLoading && !data) {
    Router.replace('/');
  }

  if (routeProtected && !isLoading && isAdmin && data?.role !== 'ADMIN') {
    Router.replace('/');
  }

  return (
    <UserContext.Provider
      value={{
        isLoading,
        user: data,
      }}
    >
      {children}
    </UserContext.Provider>
  );
}
```

and I have this for logOut

```javascript
  const logOutMutation = useMutation(() => axios.post('/api/auth/logout'), {
    onSuccess: () => {
      queryClient.clear();
    },
  });
```

When I do `logOutMutation.mutate()` I expect the console.log in the provider will fire again with undefined

**Reply by TkDodo** [reply]
`clear` doesn't re-render - `reset` does

**Reply by carlosriveroib** [reply]
And it is possible to reset all queries at once?

**Reply by carlosriveroib** [reply]
Ok, resetQueries() without any key I guess reset all, thank you very much. You are the best!

**Reply by ArturasTv** [reply]
I think you should invalidate queries on sign-in, not on logout, because if your access token expires, you will be logouted in the background, and after signing in with a different user, you will get unexpected results.

**Comment by muhammadsaqib95** [comment]
[query-invalidation](https://tanstack.com/query/v4/docs/guides/query-invalidation) This link helped me for this issue.

**Reply by Coding-Algorithm** [reply]
I think that link is not active anymore, this is active as of today [`query_invalidation`](https://tanstack.com/query/v4/docs/framework/react/guides/query-invalidation)

**Comment by obsd** [comment, marked as answer]
For all the lazy devs out there :)
```
queryClient.invalidateQueries()
```
does the magic

**Comment by Sunhat** [comment]
@obsd `resetQueries`, `invalidateQueries`, `clear`, all do different things.

According to the documentation, `invalidateQueries` marks the existing data as stale. This will trigger refetching, not initial fetching. Before calling people lazy, maybe consider understanding not everyone is looking for that.

https://tanstack.com/query/v4/docs/react/guides/query-invalidation

**Reply by obsd** [reply]
This is what the question is about, invalidation data, isn't it?

**Reply by osmanbaskaya** [reply]
Hi @obsd, I know that your intention was good. If you read the OP's question (not just the title), even though he was saying `invalidation,` you will notice that he wants to `clear` the cache to avoid leaking the information from previously signed-in users, so invalidating the queries won't be enough as it will show them before fetching the fresh ones (which they will fail because of unauthorized access).
